Mathlib/Topology/Category/Profinite/Nobeling.lean

```diff
@@ -0,0 +1,1556 @@
+  ext i
+  dsimp [SwapTrue, Proj]
+  split_ifs with hi h
+  · rw [ord_term ho] at hi
+    rw [← hx.2, hi]
+  · rfl
+  · simp only [not_lt] at h
+    have h' : o < ord I i := lt_of_le_of_ne h (Ne.symm hi)
+    specialize hsC x hx.1 i
+    rw [← not_imp_not] at hsC
+    simp only [not_lt, Bool.not_eq_true, Order.succ_le_iff] at hsC
+    exact (hsC h').symm
+
+open Classical in
+theorem CC_exact {f : LocallyConstant C ℤ} (hf : Linear_CC' C hsC ho f = 0) :
+    ∃ y, πs C o y = f := by
+  dsimp [Linear_CC', Linear_CC'₀, Linear_CC'₁] at hf
+  simp only [sub_eq_zero, ← LocallyConstant.coe_inj, LocallyConstant.coe_comap,
+    continuous_CC'₀, continuous_CC'₁] at hf
+  let C₀C : C0 C ho → C := fun x ↦ ⟨x.val, x.prop.1⟩
+  have h₀ : Continuous C₀C := Continuous.subtype_mk continuous_induced_dom _
+  let C₁C : π (C1 C ho) (ord I · < o) → C :=
+    fun x ↦ ⟨SwapTrue o x.val, (swapTrue_mem_C1 C hsC ho x).1⟩
+  have h₁ : Continuous C₁C := Continuous.subtype_mk
+    ((continuous_swapTrue o).comp continuous_subtype_val) _
+  refine ⟨LocallyConstant.piecewise' ?_ (isClosed_C0 C hC ho)
+      (isClosed_proj _ o (isClosed_C1 C hC ho)) (f.comap C₀C) (f.comap C₁C) ?_, ?_⟩
+  · rintro _ ⟨y, hyC, rfl⟩
+    simp only [Set.mem_union, Set.mem_setOf_eq, Set.mem_univ, iff_true]
+    rw [← union_C0C1_eq C ho] at hyC
+    refine hyC.imp (fun hyC ↦ ?_) (fun hyC ↦ ⟨y, hyC, rfl⟩)
+    rwa [C0_projOrd C hsC ho hyC]
+  · intro x hx
+    simpa only [h₀, h₁, LocallyConstant.coe_comap] using (congrFun hf ⟨x, hx⟩).symm
+  · ext ⟨x, hx⟩
+    rw [← union_C0C1_eq C ho] at hx
+    cases' hx with hx₀ hx₁
+    · have hx₀' : ProjRestrict C (ord I · < o) ⟨x, hx⟩ = x
+      · simpa only [ProjRestrict, Set.MapsTo.val_restrict_apply] using C0_projOrd C hsC ho hx₀
+      simp only [hx₀', hx₀, h₀, LocallyConstant.piecewise'_apply_left, LocallyConstant.coe_comap,
+        Function.comp_apply, πs_apply, continuous_projRestrict]
+    · have hx₁' : (ProjRestrict C (ord I · < o) ⟨x, hx⟩).val ∈ π (C1 C ho) (ord I · < o)
+      · simpa only [ProjRestrict, Set.MapsTo.val_restrict_apply] using ⟨x, hx₁, rfl⟩
+      simp only [πs_apply, continuous_projRestrict, LocallyConstant.coe_comap, Function.comp_apply,
+        hx₁', LocallyConstant.piecewise'_apply_right, h₁]
+      congr
+      exact C1_projOrd C hsC ho hx₁
+
+variable (o) in
+theorem succ_mono : CategoryTheory.Mono (ModuleCat.ofHom (πs C o)) := by
+  rw [ModuleCat.mono_iff_injective]
+  exact injective_πs _ _
+
+theorem succ_exact :
+    CategoryTheory.Exact (ModuleCat.ofHom (πs C o)) (ModuleCat.ofHom (Linear_CC' C hsC ho)) := by
+  rw [ModuleCat.exact_iff]
+  ext f
+  rw [LinearMap.mem_ker, LinearMap.mem_range]
+  refine ⟨fun ⟨y, hy⟩ ↦ ?_, fun hf ↦ ?_⟩
+  · simpa only [ModuleCat.ofHom, ← hy] using CC_comp_zero _ _ _ y
+  · exact CC_exact _ hC _ ho hf
+
+end ExactSequence
+
+section GoodProducts
+
+namespace GoodProducts
+
+/-- The `GoodProducts` in `C` that contain `o` (they necessarily start with `o`, see
```

Comment:
I do not understand this comment. If o is an arbitrary ordinal then a list containing o might not stat with o surely? What am I missing?